Lib/asyncio/tasks.py

```diff
@@ -599,6 +600,11 @@ def cancel(self):
         for child in self._children:
             if child.cancel():
                 ret = True
+        if ret:
```

Comment:
I don't follow why future cancellation should be propagated but if children are tasks nothing should be done.

Reply:
See https://bugs.python.org/issue26923 and, specifically, `Task.cancel()`: if the task is done/cancelled, we don't raise `CancelledError`. Similarly for `gather()`, if all of its tasks are already done, we don't raise it.

This seems to be a super rare edge case.

Reply:
Got it.